xfields/ibs/_equilibrium_emittance.py

```diff
@@ -241,83 +294,70 @@ def compute_emittance_evolution(
-        )
-        sigma_delta = input_sigma_delta
+    if overwrite_sigma_zeta is not None:
+        warnings.warn("'overwrite_sigma_zeta' is specified, make sure it remains consistent with 'initial_emittances'.")
+        sigma_zeta = overwrite_sigma_zeta
+    elif overwrite_sigma_delta is not None:
+        warnings.warn("'sigma_delta' is specified, make sure it remains consistent with 'initial_emittances'.")
+        sigma_delta = overwrite_sigma_delta
     longitudinal_emittance_ratio = sigma_zeta / sigma_delta
-    if input_sigma_zeta is not None or input_sigma_delta is not None:
+    if overwrite_sigma_zeta is not None or overwrite_sigma_delta is not None:
         assert initial_emittances is not None, (
-            "Input of 'input_sigma_zeta' or 'input_sigma_delta' provided, but "
+            "Input of 'overwrite_sigma_zeta' or 'overwrite_sigma_delta' provided, but "
             "not of 'initial_emittances'. Please provide 'initial_emittances'."
         )
-
-    time = []
-    emittances_x_list, emittances_y_list, emittances_z_list = [], [], []
-    T_x, T_y, T_z = [], [], []
-
+    # ----------------------------------------------------------------------------------------------
+    # Start structures to store the iterative results until convergence
+    tolerance = np.inf
     time_step = twiss.T_rev0  # Initial time step is the revolution period
-    tol = np.inf
 
+    time_deltas = []
+    res_gemitt_x, res_gemitt_y, res_gemitt_zeta = [], [], []
+    T_x, T_y, T_z = [], [], []
     current_emittances = np.array([emittance_x, emittance_y, emittance_z])
-    it = 0  # Iteration counter
-
-    while tol > rtol:
+    iterations = 0  # Iteration counter
+    # ----------------------------------------------------------------------------------------------
+    # Start the iterative process until convergence:
+    # - Compute IBS rates and emittance time derivatives
+    # - Update emittances using the time derivatives and time step
+    # - Enforce transverse / longitudinal constraints if specified
+    # - Store all intermediate results for this time step
+    # - Compute tolerance and check for convergence
+    while tolerance > rtol:
         # Print convergence progress
-        sys.stdout.write("\rConvergence = {:.1f}%".format(100 * rtol / tol))
+        sys.stdout.write(f"\rIteration {iterations} - convergence = {100 * rtol / tolerance:.1f}%")
 
         # Compute IBS growth rates and emittance derivatives
         ibs_growth_rates, emittance_derivatives = _ibs_rates_and_emittance_derivatives(
-            twiss,
-            bunch_intensity,
-            current_emittances,
-            initial_emittances=initial_emittances,
-            emittance_coupling_factor=emittance_coupling_factor,
-            emittance_constraint=emittance_constraint,
+            twiss=twiss,
+            formalism=formalism,
+            total_beam_intensity=total_beam_intensity,
+            input_emittances=current_emittances,
```

Comment:
I pass by name the arguments to make sure we know what we're doing.

I removed the `emittance_coupling_factor` and `emittance_constraint` as the private function does not use them. Previously they were considered `kwargs` which were not transmitted anywhere, so ignored. Since we now transmit `kwargs` to the IBS growth rates computation we have to remove these.